### PR TITLE
Add async catch method

### LINKED ISSUE #429: Add async support to Catch Exception
Please add support for Catch.Exception(async)

```
 Because of = async () =>
                _exception = await Catch.Exception(() => PopAsync(x)));
  
It should_throw_an_exception = () => 
                _exception.Message.ShouldBe("Outer exception, not aggregate");
```

Thanks

### PR DESCRIPTION
**Pull request by robertcoltheart**
Fixes #429

### CODE REVIEW

**Comment by mikeblakeuk** [comment]
Amazing, thank you. What version will this be in? (i assume on nuget.com  and net 462+, so release branch)

File: src/Machine.Specifications/Catch.cs [hunk 1]
```diff
@@ -1,30 +1,61 @@
...
+            {
+                return ex;
+            }
+
             return null;
         }
 
         public static TException Only<TException>(Action throwingAction)
-          where TException : Exception
+            where TException : Exception
```
Comment (mikeblakeuk) [inline src/Machine.Specifications/Catch.cs:58]: Should Only also be OnlyAsync ?

**Comment by robertcoltheart** [comment]
This has been released to Nuget now.

File: src/Machine.Specifications/Catch.cs (hunk 1 above)
Comment (robertcoltheart) [reply src/Machine.Specifications/Catch.cs:58]: No, I'm going to be deprecating the `Only` method and removing it entirely in a later major version. It breaks the AAA pattern. Will write a longer post when I make the change. But see here for starters: https://github.com/xunit/xunit/issues/188#issuecomment-104378832

**Pull request merged**